src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,71 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
+) {
+	std::map<PopType const* const, fixed_point_t> employees_per_job {};
+	for (const Job job : production_type.get_jobs()) {
+		PopType const* const pop_type = job.get_pop_type();
+		employees_per_job[pop_type] = 0;
+	}
+
+	for (std::pair<Pop const* const, const fixed_point_t> pop_and_number_employed : employees) {
+		Pop const* const pop = pop_and_number_employed.first;
+		const fixed_point_t number_employed = pop_and_number_employed.second;
+		PopType const* const pop_type = &(pop->get_type());
+		employees_per_job[pop_type] += number_employed;
+	}
+
+	fixed_point_t my_input_multiplier = input_modifier, my_throughput_multiplier = throughput_modifier,
+				  my_output_multiplier = output_modifier;
```

Comment:
These are assumed to include the base `1.0` / `100%`, does that come from any value in the defines (e.g. a default input modifier of `100%`) or will it need to be used as the base when calculating the sum of exernal modifiers?

Reply:
As mentioned in a different comment, the parameters of the produce method comes from the state. The state will have a similar method that gets its parameters from a method in the country. The country has to handle the `1 + modifiers`.